components/server-side-render/index.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies.
+ */
+import { isEqual, isObject, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: null,
+		};
+	}
+
+	componentDidMount() {
+		this.fetch( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps, this.props ) ) {
+			this.fetch( nextProps );
+		}
+	}
+
+	fetch( props ) {
+		this.setState( { response: null } );
+		const { block, attributes } = props;
+
+		const path = '/gutenberg/v1/block-renderer/' + block + '?context=edit&' + this.getQueryUrlFromObject( { attributes } );
+
+		return wp.apiRequest( { path: path } ).then( ( response ) => {
+			if ( response && response.rendered ) {
+				this.setState( { response: response.rendered } );
+			}
+		} );
+	}
+
+	getQueryUrlFromObject( obj, prefix ) {
```

Comment:
This sounds like a `wordpress/url` package utility.

Reply:
>This sounds like a wordpress/url package utility.

In fact, it could probably be built-in as the default behavior for the existing [`addQueryArgs` function](https://github.com/WordPress/packages/blob/ee842ec2c030ee6c0c917e451a79d5f9bd931b28/packages/url/src/index.js#L10-L24).

Reply:
`addQueryArgs` works well but not for nested attributes, thus the need for creating a custom method. Couldn't find an existing utility that would put together URL in case of nested params, if you happen to know any, happy to change it!

Reply:
If the behavior is needed, it could be added to the existing function via pull request to the [packages repository](https://github.com/WordPress/packages).